### PR TITLE
Hbos

### LINKED ISSUE #1521: [Idea]: <HBOS>
### Idea Title

Histogram-Based Outlier Score (HBOS)

### Idea Description

Histogram-Based Outlier Score (HBOS) is an unsupervised anomaly detection technique that assumes independence between features and analyzes each feature independently. It calculates histograms for each feature and then evaluates the "outlierness" of a data point based on its position in the histogram bins.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [ ] Adds a new feature
- [ ] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by ananyag309**
### Description
Added the document

Fixes #1521 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ananyag309** [comment]
@ajay-dhangar 
may i know what the issue is so that i can correct it?

File: docs/machine-learning/HBOS.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+### Histogram-Based Outlier Score (HBOS)
+
```
Comment (ajay-dhangar) [inline docs/machine-learning/HBOS.md]: ⁉️

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
follow our format

**Comment by ananyag309** [comment]
@ajay-dhangar please check now ... any further changes need please let me know

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**